stage0/src/sev.rs

```diff
@@ -205,12 +205,32 @@ pub fn share_page(page: Page<Size4KiB>) {
 
 /// Stops sharing a single 4KiB page with the hypervisor when running with AMD SEV-SNP enabled.
 pub fn unshare_page(page: Page<Size4KiB>) {
+    let page_start = page.start_address().as_u64();
+    // Only the first 2MiB is mapped as 4KiB pages, so make sure we fall in that range.
+    assert!(page_start < Size2MiB::SIZE);
     if sev_status().contains(SevStatus::SNP_ACTIVE) {
         let page_start = page.start_address().as_u64();
```

Comment:
Drop this `let` (although the compiler should be smart enough to figure it out).

Reply:
The joys of copying and pasting. Done.